documentation/swagger-authoring-automated-guidelines.md

```diff
@@ -0,0 +1,96 @@
+# Automated Guidelines for OpenAPI(swagger) specs Authoring #
+
+This document lists the set of automated rules that can be validated against OpenAPI(swagger) spec by running [validation tools](https://github.com/Azure/adx-documentation-pr/wiki/OpenAPI-Validation-tools). Please look [here for Manual guidelines](swagger-authoring-manual-guidelines.md).
+
+It is a requirement to conform to all manual and automated rules with severity "Error" before sending a pull request for review.
+
+We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an intimate knowledge of a service endpoint and its developer experience to avoid feeding inefficiencies into teams that focus on Azure developer experiences and the rest of the Azure eco-system.
+
+## Error vs Warning
+- Rules with severity "Error" have to be addressed for the OpenAPI(swagger) spec to be approved by the reviewers. If there is a strong reason for why the rule cannot be addressed in an OpenAPI(swagger) spec, [suppression process](https://github.com/Azure/adx-documentation-pr/wiki/Swagger-Validation-Errors-Suppression) has to be followed to get sign off.
+
+- Rules with severity "Warning" are either strong recommendations made by Azure developer experience team for better SDK/Documentation experience or they point out something to evaluate, for example, the warning for booleans asks the user to evaluate whether the property should be a boolean or not. It is NOT a requirement to address warnings.
```

Comment:
How about if we say 
> It is highly recommended to address warnings to have best of quality SDKs + Documentations. 

instead of

> It is NOT a requirement to address warnings

Personally, I just want to push the team to address the warnings as well but do not want to say `NOT a requirement`. Thoughts/Opinions?